src/debug/shared/dbgtransportsession.cpp

```diff
@@ -1344,8 +1349,7 @@ void DbgTransportSession::TransportWorker()
             eStatus = SCS_NetworkFailure;
         else
         {
-            DWORD pid = GetCurrentProcessId(); 
-            if (m_pipe.CreateServer(pid) && m_pipe.WaitForConnection())
```

Comment:
This transport thread operates in a retry loop and when it retries it will first call m_pipe.Disconnect() before continuing back to line 1307. That will transition the TwoWayPipe into the NotInitialized state. From there you must call CreateServer() to get back to the Created state where WaitForConnection() works. I think you want something like:

    if( (m_pipe.GetState() == Created || m_pipe.CreateServer(pid) &&
         m_pipe.WaitForConnection())
    {
          ...
    }